deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/WritePathAttributeHandler.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.server.deployment.scanner;
+
+import static org.jboss.as.server.deployment.scanner.logging.DeploymentScannerLogger.ROOT_LOGGER;
+
+import java.io.File;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.ReloadRequiredWriteAttributeHandler;
+import org.jboss.as.controller.OperationContext.Stage;
+import org.jboss.as.controller.services.path.PathManager;
+import org.jboss.dmr.ModelNode;
+
+/**
+ * @author wangc
+ *
+ */
+public class WritePathAttributeHandler extends ReloadRequiredWriteAttributeHandler {
+
+    private final PathManager pathManager;
+
+    public WritePathAttributeHandler(PathManager pathManager) {
+        super(DeploymentScannerDefinition.PATH);
+        this.pathManager = pathManager;
+    }
+
+    @Override
+    protected boolean applyUpdateToRuntime(OperationContext context, ModelNode operation, String attributeName,
+            ModelNode resolvedValue, ModelNode currentValue, HandbackHolder<Void> voidHandback)
+                    throws OperationFailedException {
+        boolean reloadRequired = super.applyUpdateToRuntime(context, operation, attributeName, resolvedValue, currentValue,
+                voidHandback);
+        if (reloadRequired) {
+            final ModelNode model = context.readResource(PathAddress.EMPTY_ADDRESS).getModel();
+            final ModelNode relativeTo = DeploymentScannerDefinition.RELATIVE_TO.resolveModelAttribute(context, model);
+            // The value actually changed so we need to add step to validate the path
+            context.addStep(new OperationStepHandler() {
+                @Override
+                public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                    File fullPath;
```

Comment:
In this specific case I think we should be using `java.nio.file.Path`. The `fullPath.canWrite()` is not reliable per the JavaDoc while `java.nio.file.Files.isReadable()` tends to be more reliable.

> Tests whether the application can read the file denoted by this abstract pathname. On some platforms it may be possible to start the Java virtual machine with special privileges that allow it to read files that are marked as unreadable. Consequently this method may return true even though the file does not have read permissions.